pkg/controller/gitopsconfig/controller.go

```diff
@@ -491,43 +487,52 @@ func (job *jobdata) watchjob(instance *gitopsv1alpha1.GitOpsConfig) {
 		if err != nil {
 			continue
 		}
-		instance.Status = jobState
+
 		if jobState.State != "INPROGRESS" {
-			instance.Status.EndTime = jobState.EndTime
-			instance.Status.State = jobState.State
-			err := job.updateStatus(instance)
+			err := job.updateStatus(instance, jobState)
 			if err != nil {
-				log.Info("Error in Updating the job when jobState.Active == 0:" + err.Error())
+				log.Error(err, "Error in Updating the job when jobState.Active == 0")
 			}
 			return
 		}
 
 		if counter == 0 {
-			instance.Status.State = jobState.State
-			instance.Status.EndTime = ""
+			jobState.CompletionTime = nil
 			log.Info("Intial Update")
-			err := job.updateStatus(instance)
+			err := job.updateStatus(instance, jobState)
 			if err != nil {
-				log.Info("Error in Intial Update of job Status:" + err.Error())
+				log.Error(err, "Error in Intial Update of job Status:")
 			}
 		}
 
 		counter++
 	}
 }
 
-func (job *jobdata) updateStatus(instance *gitopsv1alpha1.GitOpsConfig) error {
-
-	err := job.Reconcile.client.Status().Update(context.TODO(), instance)
+func (job *jobdata) updateStatus(instance *gitopsv1alpha1.GitOpsConfig, jobState gitopsv1alpha1.GitOpsConfigStatus) error {
```

Comment:
if we go with `instance.Name` and `instance.Namespace` .. getting the below error 
`{"level":"error","ts":1575012770.129529,"logger":"controller_gitopsconfig","msg":"Error in gettting the instance","error":"GitOpsConfig.eunomia.kohls.io \"hello-world-helm\" not found"`.. even though `instance` is a pointer it will be only 4 bytes